### PR TITLE
feat: add date control type

### PR DESCRIPTION
**Pull request by dannyhw**
Issue: #199 

## What I did

I added the date control type

I also updated the date picker library to resolve the yellow warnings it triggered previously and fix the web version of this control.

## How to test

First make sure you run yarn install or yarn bootstrap in the root and then run pod install in the example app

- Open the example app
- open the date story
- Adjust the date by using the control in the addons tab

### CODE REVIEW

File: addons/ondevice-controls/src/types/Date.tsx [hunk 1]
```diff
@@ -23,82 +23,66 @@ const Label = styled.Text(({ theme }: any) => ({
...
+    setIsDateVisible(false);
+    setIsTimeVisible(false);
   };
 
-  onDatePicked = (date) => {
-    const value = date.valueOf();
-    const { onChange } = this.props;
-    onChange(value);
-    this.hidePicker();
+  const onDatePicked = (pickedDate) => {
```
Comment (lauriharpf) [inline addons/ondevice-controls/src/types/Date.tsx]: Very minor, but perhaps we could type this and remove the commented-out line below 🤔 

```suggestion
  const onDatePicked = (pickedDate: Date) => {
```

File: addons/ondevice-controls/src/types/Date.tsx [hunk 2]
```diff
@@ -23,82 +23,66 @@ const Label = styled.Text(({ theme }: any) => ({
...
-
-  constructor(props) {
-    super(props);
-    this.state = {
-      isDateVisible: false,
-      isTimeVisible: false,
-    };
-  }
+const DateType = ({ onChange, arg: { name, value } }: DateProps) => {
+  const [isDateVisible, setIsDateVisible] = useState(false);
```
Comment (lauriharpf) [inline addons/ondevice-controls/src/types/Date.tsx]: This split to two different boolean variables (`isDateVisible` and `isTimeVisible`) pre-dates your code, but I'm tempted to suggest a small refactor here: https://github.com/storybookjs/react-native/pull/225

It's by no means a priority or a blocker, but thought you might be able to take a quick peek now that this code is fresh in the minds of both of us.

File: examples/native/components/ControlExamples/Date/Date.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { Text } from 'react-native';
+
+interface DateProps {
+  date: Date;
+}
+
+export const LocalisedDate = ({ date }: DateProps) => <Text>{date.toLocaleDateString()}</Text>;
```
Comment (lauriharpf) [inline examples/native/components/ControlExamples/Date/Date.tsx]: Could we also show the selected time here, not just the date?

```suggestion
export const LocalisedDate = ({ date }: DateProps) => (
  <Text>
    {date.toLocaleDateString()} {date.toLocaleTimeString()}
  </Text>
);
```

**Review by lauriharpf — APPROVED** [review]
Again superb work! 🚀 💪 💎 

Some minor comments, take a peek if you have the time for it. In general the control seems to work very well and the conversion from the knob looks smooth :+1:.

File: addons/ondevice-controls/src/types/Date.tsx (hunk 2 above)
Comment (dannyhw) [reply addons/ondevice-controls/src/types/Date.tsx]: accepted your PR, seems like a great idea to me. Thanks again!

**Pull request merged**